src/NServiceBus.Core/Transports/TransportSendingConfigurationResult.cs

```diff
@@ -0,0 +1,26 @@
+namespace NServiceBus.Transports
+{
+    using System;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Represents the result for configuring the transport for sending.
+    /// </summary>
+    public class TransportSendingConfigurationResult
+    {
+        /// <summary>
+        /// Creates new result object.
+        /// </summary>
+        public TransportSendingConfigurationResult(Func<IDispatchMessages> dispatcherFactory,
+            Func<Task> preStartupCheck)
```

Comment:
The problem with `bool` is that we would not be able to return the actual cause of the failure. So we would have to have a struct containing a bool and a string for the message. Or ideally something like

```
return CheckResult.OK;
//or
return CheckResult.Error("put error message here")
```

Thoughts @danielmarbach @tmasternak ?